parser.go

```
@@ -99,6 +104,52 @@ func (amg *AuditMessageGroup) AddMessage(am *AuditMessage) {
 	}
 }
 
+// Find all `saddr=` occurrences in a message and do a lookup
+func (amg *AuditMessageGroup) mapDns(am *AuditMessage) {
+	data := am.Data
+	start := 0
+	end := 0
+
+	for {
+		if start = strings.Index(data, "saddr="); start < 0 {
+			break
+		}
+
+		// Progress the start point beyond the = sign
+		start += 6
+		if end = strings.IndexByte(data[start:], spaceChar); end < 0 {
+			end = len(data) - start
+			if end > 34 {
+				break
+			}
+		}
+
+		saddr := data[start : start+end]
+
+		var ip string
+
+		switch family := saddr[0:4]; family {
+		// 0200: ipv4
+		case "0200":
+			octet, _ := hex.DecodeString(saddr[8:16])
```

Don't ignore the error